rotkehlchen/chain/ethereum/adex/adex.py

```diff
@@ -0,0 +1,780 @@
+                    from_timestamp=from_timestamp,
+                    to_timestamp=to_timestamp,
+                )
+            except RemoteError:
+                return []
+
+            all_events.extend(events)
+
+        for address in addresses:
+            self.database.update_used_query_range(
+                name=f'{ADEX_EVENTS_PREFIX}_{address}',
+                start_ts=from_timestamp,
+                end_ts=to_timestamp,
+            )
+        return all_events
+
+    @staticmethod
+    def _get_user_identity(address: ChecksumAddress) -> ChecksumEthAddress:
+        """Given an address (signer) returns its protocol user identity.
+        """
+        return generate_address_via_create2(
+            address=HexAddress(HexStr(IDENTITY_FACTORY_ADDR)),
+            salt=HexStr(CREATE2_SALT),
+            init_code=HexStr(IDENTITY_PROXY_INIT_CODE.format(signer_address=address)),
+        )
+
+    def get_balances(
+            self,
+            addresses: List[ChecksumAddress],
+    ) -> Dict[ChecksumAddress, List[ADXStakingBalance]]:
+        """Return the addresses' balances (staked amount per pool) in the AdEx
+        protocol.
+
+        TODO: route non-premium users through on-chain query.
+        """
+        staking_balances: Dict[ChecksumAddress, List[ADXStakingBalance]] = {}
+        is_graph_mode = self.graph and self.premium
+        if is_graph_mode:
+            try:
+                bonds = self._get_staking_events_graph(
+                    addresses=addresses,
+                    event_type=EventType.BOND,
+                )
+                unbonds = self._get_staking_events_graph(
+                    addresses=addresses,
+                    event_type=EventType.UNBOND,
+                )
+                unbond_requests = self._get_staking_events_graph(
+                    addresses=addresses,
+                    event_type=EventType.UNBOND_REQUEST,
+                )
+            except RemoteError:
+                return staking_balances
+
+            adx_usd_price = Inquirer().find_usd_price(EthereumToken('ADX'))
+            staking_balances = self._calculate_staking_balances(
+                bonds=bonds,
+                unbonds=unbonds,
+                unbond_requests=unbond_requests,
+                adx_usd_price=adx_usd_price,
+            )
+        else:
+            raise NotImplementedError(
+                "Get AdEx balances for non premium user is not implemented.",
+            )
+        return staking_balances
+
+    def get_events_history(
+            self,
+            addresses: List[ChecksumEthAddress],
+            reset_db_data: bool,
+            from_timestamp: Timestamp,
+            to_timestamp: Timestamp,
+    ) -> Dict[ChecksumAddress, ADXStakingHistory]:
+        """Get the staking history events of the addresses in the AdEx protocol.
+        """
+        if self.graph is None:  # could not initialize graph
+            return {}
+
+        with self.trades_lock:
```

Comment:
What's this lock for? I only see it in one place. In that case it's useless, no?